3rdParty/fuerte/src/HttpConnection.cpp

```diff
@@ -350,28 +377,37 @@ template <SocketType ST>
 void HttpConnection<ST>::asyncWriteNextRequest() {
   FUERTE_LOG_HTTPTRACE << "asyncWriteNextRequest: this=" << this << "\n";
   assert(_active.load());
+  assert(_item == nullptr);
 
   http::RequestItem* ptr = nullptr;
   if (!_queue.pop(ptr)) {
     _active.store(false);
-    if (!_queue.pop(ptr)) {
+    if (_queue.empty()) {
       FUERTE_LOG_HTTPTRACE << "asyncWriteNextRequest: stopped writing, this="
                            << this << "\n";
       if (_shouldKeepAlive && this->_config._idleTimeout.count() > 0) {
         FUERTE_LOG_HTTPTRACE << "setting idle keep alive timer, this=" << this
                              << "\n";
-        setTimeout(this->_config._idleTimeout);
+        int exp = 2;
+        _leased.compare_exchange_strong(exp, 0);
```

Comment:
If the value is already 0, then there is no need to do anything. If the value is still 1, then we must not do anything here, since the connection was leased, but no `sendRequest` has happened yet.

Reply:
Comment added.